src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  let newState = state;
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newState = Object.assign(newState, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of Object.values(action.keysToRemove)) {
+        delete newState[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const key in newState) {
+        delete newState[key];
+      }
+    }
+  }
+
+  return newState;
```

Comment:
You don't need to return newState, as the function is supposed to modify the given state object directly.